src/moma.cpp

```diff
@@ -85,20 +90,198 @@ MoMA::MoMA(const arma::mat &i_X,  // Pass X_ as a reference to avoid copy
     is_solved      = false;  // TODO: check if alphauv == 0
 };
 
-int MoMA::deflate(double d)
+// Initializer for LDA and CCA
+MoMA::MoMA(
+    // const arma::mat &X_,  // Pass X_ as a reference to avoid copy
+    const arma::mat &i_X_working,
+    const arma::mat &i_Y_working,
+    /*
+     * sparsity - enforced through penalties
+     */
+    double i_lambda_u,  // regularization level
+    double i_lambda_v,
+    Rcpp::List i_prox_arg_list_u,
+    Rcpp::List i_prox_arg_list_v,
+
+    /*
+     * smoothness - enforced through constraints
+     */
+    double i_alpha_u,  // Smoothing levels
+    double i_alpha_v,
+    const arma::mat &i_Omega_u,  // Smoothing matrices
+    const arma::mat &i_Omega_v,
+
+    /*
+     * Algorithm parameters:
+     */
+    double i_EPS,
+    long i_MAX_ITER,
+    double i_EPS_inner,
+    long i_MAX_ITER_inner,
+    std::string i_solver,
+    DeflationScheme i_ds)
+    : MoMA(i_X_working.t() * i_Y_working,  // the matrix on which we find pSVD
+           i_lambda_u,
```

Comment:
Here is why `X_original` cannot be zero.

Reply:
What is this a reference to? Did I ask about zeros?

Reply:
Oh sorry, it should be "here is why X_original cannot be constant".